src/backend.cpp

```diff
@@ -1,138 +1,138 @@
+#include <QDebug>
+#include <QDBusConnection>
 #include "backend.h"
+#include "ocihelperadaptor.h"	//generated during build time
 
-void Backend::install()
-{
-    addRepositories();
-
-    system( "zypper refresh" );
-
-    QString zypper_command = "2>&1 zypper -x -n in";
-    foreach( QString package, packages() ) {
-        zypper_command.append( QString( " %1" ).arg( package ) );
+Backend::Backend()
+{	
+    new OCIHelperAdaptor( this );
+    QDBusConnection connection = QDBusConnection::systemBus();
+    if ( !connection.isConnected() ) {
+	qFatal( "Cannot connect to the D-Bus system bus" );
+	exit( 1 );
     }
-
-    m_adaptor = new DBusAdaptor( this );
-    QDBusConnection connection = QDBusConnection::sessionBus();
-    connection.registerService( "org.opensuse.oneclickinstaller" );
     connection.registerObject( "/", this );
+    if ( !connection.registerService( "org.opensuse.OCIHelper" ) ) {
+	qDebug() << qPrintable( connection.lastError().message() );
+	exit( 1 );
+    }
 
-    zypper_command.append( " | grep -o --line-buffered \"percent=\\\"[0-9]*\\\"\" | " );
-    zypper_command.append( "while read line; do echo \"$line\" > /var/log/oneclick.log; echo \"$line\" > /var/log/oneclick.log; done" );
-
-    QProcess::execute( zypper_command.toStdString().c_str() );
-}
-
-Backend::Backend()
-{
-    m_manager = new zypp::RepoManager;
-    m_ptr = new zypp::ZYpp::Ptr;
-    m_keyRingManager = new zypp::KeyRingCallbacks;
-    m_errorCode = 0;
-}
-
-void Backend::addRepositories()
-{
-    qDebug() << "add repos called";
-
-    foreach( QUrl iter, repositories() ) {
-        zypp::RepoInfo repoInfo;
-        std::string url = iter.toString().toStdString();
-        repoInfo.addBaseUrl( zypp::Url( url ) );
-        repoInfo.setAlias( url );
-        repoInfo.setGpgCheck( true );
-
-        zypp::KeyRing::setDefaultAccept( zypp::KeyRing::TRUST_AND_IMPORT_KEY );
-
-        if( !exists( QString::fromStdString( url ) ) ) {
-            try {
-                m_manager->addRepository( repoInfo );
-            } catch( zypp::repo::RepoMetadataException e ) {
-                m_errorCode = 4;
-            }
-            catch( zypp::repo::RepoException ) {
-                m_errorCode = 5;
-            }
-        }
-        else
-            qDebug() << "Repository Exists";
-//        m_manager->buildCache( repoInfo );
-//        m_manager->loadFromCache( repoInfo );
+    // Proxy for OCI
+    QDBusConnection sysBus = QDBusConnection::systemBus();
+    if ( !sysBus.isConnected() ) {
+	qFatal( "Cannot connect to the D-Bus system bus" );
+	exit( 1 );
+    }
+    m_oci = new org::opensuse::oneclickinstaller( "org.opensuse.oneclickinstaller", "/OCI", sysBus, this );
+    if ( !m_oci->isValid() ) {
+	qFatal( "Oops! Cannot connect to the service org.opensuse.oneclickinstaller" );
+	exit( 1 );
     }
+    
+    /******************************************* INSTALLATION *******************************************/
+    // Step 1. Mark packages for installation
```

Comment:
Please put here a "// Fixme: hardcoded installation package"  comment (or something like that)